creativecloud/scripts/mep/aexg4866/firefly/firefly-masonry.js

```diff
@@ -0,0 +1,216 @@
+// import loadStyle for utils not working with replacepage
+import { getLibs, getConfig } from '../../../utils.js';
+
+const { createTag, createIntersectionObserver } = await import(`${getLibs()}/utils/utils.js`);
+const { focusOnInput } = await import('../../../../features/firefly/firefly-interactive.js');
+const { createPromptField, createEnticement } = await import('../../../../features/interactive-elements/interactive-elements.js');
+
+function addKeyEvent(promptInput, promptButton) {
+  promptInput.addEventListener('keydown', (event) => {
+    if (event.keyCode === 13) {
+      event.preventDefault();
+      promptButton.click();
+    }
+    if (event.target.value.length === 0 && event.keyCode === 32) {
+      if (event.preventDefault) event.preventDefault();
+      else event.returnValue = false;
+    }
+  });
+}
+
+function handleTouchDevice(mediaContainer, delay) {
+  let tapCount = 0;
+  const aTag = mediaContainer.querySelector('a');
+  const imgContent = mediaContainer.querySelector('.image-content');
+  aTag.addEventListener('touchstart', (e) => {
+    e.preventDefault();
+    tapCount += 1;
+    if (tapCount === 2) aTag.click();
+    imgContent.style.opacity = 1;
+    setTimeout(() => {
+      tapCount = 0;
+      imgContent.style.opacity = 0;
+    }, delay);
+  });
+}
+
+function getImgSrc(pic, viewport = '') {
+  let source = '';
+  if (viewport === 'mobile') source = pic.querySelector('source[type="image/webp"]:not([media])');
+  else source = pic.querySelector('source[type="image/webp"][media]');
+  return source.srcset;
+}
+
+async function createEmbellishment(allP, media, mediaMobile, ic, mode, interactiveMode, el) {
+  const [promptText, buttonText] = allP[3].innerText.split('|');
+  const fireflyPrompt = await createPromptField(`${promptText}`, `${buttonText}`, `ff-masonry, ${interactiveMode}`);
+  fireflyPrompt.classList.add('ff-masonry-prompt');
+  const enticementText = allP[0].textContent.trim();
+  const enticementIcon = allP[0].querySelector('a').href;
+  const enticementDiv = await createEnticement(`${enticementText}|${enticementIcon}`, mode);
+  const mediaDiv = [mediaMobile, media];
+  mediaDiv.forEach((div) => {
+    div.appendChild(fireflyPrompt.cloneNode(true));
+    div.appendChild(enticementDiv.cloneNode(true));
+    const promptButton = div.querySelector('.masonry-generate');
+    const promptInput = div.querySelector('.masonry-prompttext');
+    promptButton.addEventListener('click', async (e) => {
```

Comment:
I don't believe this click event listener contains any asynchronous operations that would require async